src/components/ListItem.tsx

```diff
@@ -29,6 +29,36 @@ export function ListItemCheckBox({ item, listPath }: Props) {
 				? !moreThan24HoursPassed(item.dateLastPurchased.toDate())
 				: false;
 
+	const getUrgencyStatus = (item: ListItem) => {
+		const currentDate = new Date();
+		const daysUntilNextPurchase = getDaysBetweenDates(
+			new Date(),
+			item.dateNextPurchased.toDate(),
+		);
+
+		const daysSinceItemLastActivity = item.dateLastPurchased
+			? getDaysBetweenDates(currentDate, item.dateLastPurchased.toDate())
+			: getDaysBetweenDates(currentDate, item.dateCreated.toDate());
+
+		if (currentDate > item.dateNextPurchased.toDate()) {
```

Comment:
_tiny_ suggestion that's damn near a nit pick

something I encourage is thinking a lot about where you initialize variables/perform logic inside "groupings" in logically a reasonable order. This function violates that a bit: `daysUntilNextPurchase` is calculated at the top, but it isn't' used until line `51` - this could mean that we're doing an operation that _might_ have some performance cost that we never needed to "spend" (I don't think it's a big deal here, just something to consider generally).

Things like this also make code a bit harder to read. When reviewing logic, it's convenient to glance up a few lines (when possible) and see the value calculated that's then being compared in `if`s

`daysSinceItemLastActivity` is another example of this. It might not ever be used if line 44 returns